src/lib/Tests/RepositoryForms/Form/Processor/PreviewFormProcessorTest.php

```diff
@@ -0,0 +1,290 @@
+use EzSystems\RepositoryForms\Data\Content\ContentCreateData;
+use EzSystems\RepositoryForms\Event\FormActionEvent;
+use Symfony\Component\Form\FormConfigInterface;
+use Symfony\Component\HttpFoundation\RedirectResponse;
+use Symfony\Component\Routing\Generator\UrlGeneratorInterface;
+use Symfony\Component\Translation\TranslatorInterface;
+use Symfony\Component\Form\FormInterface;
+use eZ\Publish\Core\Repository\Values\ContentType\FieldDefinition;
+use eZ\Publish\API\Repository\Values\Content\LocationCreateStruct;
+use eZ\Publish\Core\Repository\Values\Content\Content;
+use eZ\Publish\API\Repository\Values\Content\Content as APIContent;
+use PHPUnit\Framework\MockObject\MockObject;
+use eZ\Publish\Core\Repository\Values\ContentType\ContentType;
+
+class PreviewFormProcessorTest extends TestCase
+{
+    /** @var ContentService $contentService */
+    private $contentService;
+
+    /** @var UrlGeneratorInterface $urlGenerator */
+    private $urlGenerator;
+
+    /** @var NotificationHandlerInterface $notificationHandler */
+    private $notificationHandler;
+
+    /** @var TranslatorInterface $translator */
+    private $translator;
+
+    public function setUp()
+    {
+        $this->contentService = $this->createMock(ContentService::class);
+        $this->urlGenerator = $this->createMock(UrlGeneratorInterface::class);
+        $this->notificationHandler = $this->createMock(NotificationHandlerInterface::class);
+        $this->translator = $this->createMock(TranslatorInterface::class);
+    }
+
+    /**
+     * @param ContentService|null $contentService
+     * @param UrlGeneratorInterface|null $urlGenerator
+     * @param NotificationHandlerInterface|null $notificationHandler
+     * @param TranslatorInterface|null $translator
+     *
+     * @return PreviewFormProcessor
+     */
+    private function createPreviewFormProcessor(
+        ContentService $contentService = null,
+        UrlGeneratorInterface $urlGenerator = null,
+        NotificationHandlerInterface $notificationHandler = null,
+        TranslatorInterface $translator = null
+    ): PreviewFormProcessor {
+        return new PreviewFormProcessor(
+            $contentService ?? $this->contentService,
+            $urlGenerator ?? $this->urlGenerator,
+            $notificationHandler ?? $this->notificationHandler,
+            $translator ?? $this->translator
+        );
+    }
+
+    public function testProcessPreview()
+    {
+        $languageCode = 'cyb-CY';
+        $contentDraftId = 123;
+        $url = 'http://url';
+        $fieldDefinitionIdentifier = 'identifier_1';
+        $fieldDataValue = 'some_value';
+
+        /**
+         * $data variable in PreviewFormProcessor classs.
+         */
+        $contentStruct = $this->generateContentStruct($languageCode, $fieldDefinitionIdentifier, $fieldDataValue);
+
+        $contentDraft = $this->generateContentDraft($contentDraftId, $languageCode);
+
```

Comment:
Can you work on code style a bit more? I feel like separating everything with a new line doesn't make sense here.